dialogue-core/src/main/java/com/palantir/dialogue/core/Uris.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.core;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import com.palantir.dialogue.DialogueImmutablesStyle;
+import com.palantir.logsafe.Arg;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.Unsafe;
+import com.palantir.logsafe.UnsafeArg;
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import com.palantir.logsafe.logger.SafeLogger;
+import com.palantir.logsafe.logger.SafeLoggerFactory;
+import com.palantir.tritium.metrics.caffeine.CacheStats;
+import com.palantir.tritium.metrics.registry.SharedTaggedMetricRegistries;
+import java.net.URI;
+import java.time.Duration;
+import javax.annotation.Nullable;
+import org.immutables.value.Value;
+
+public final class Uris {
+    private static final SafeLogger log = SafeLoggerFactory.get(Uris.class);
+
+    /**
+     * This prefix may reconfigure several aspects of the client to work better in a world where requests are routed
+     * through a service mesh like istio/envoy.
+     */
+    private static final String MESH_PREFIX = "mesh-";
```

Comment:
At this point the `mesh-` prefix is no longer used (or supported) in any environment -- it should make things simpler if we introduce another change first to remove support

Reply:
I took a look at removing mesh mode, and it has tentacles across a bunch of modules & tests, so might hold off on removing that until after this PR if that's ok.